### PR TITLE
[utils] Work around CLOCK_MONOTONIC being defined on macOS 10.12 SDK despite targeting earlier releases

### PR DESCRIPTION
**Pull request by akoeplinger**
Should fix https://bugzilla.xamarin.com/show_bug.cgi?id=41786

/cc @rolfbjarne

### CODE REVIEW

**Comment by vargaz** [comment]
LGTM

**Comment by rolfbjarne** [comment]
I can confirm this fixes the problem.

File: mono/utils/mono-time.c [hunk 1]
```diff
@@ -142,7 +142,16 @@ gint64
 mono_100ns_ticks (void)
 {
 	struct timeval tv;
-#ifdef CLOCK_MONOTONIC
+#if defined(PLATFORM_MACOSX)
+	/* http://developer.apple.com/library/mac/#qa/qa1398/_index.html */
+	static mach_timebase_info_data_t timebase;
+	guint64 now = mach_absolute_time ();
```
Comment (evincarofautumn) [inline mono/utils/mono-time.c]: Out of curiosity, does it matter for this function that `mach_absolute_time ()` stops updating when the device goes to sleep?

Comment (akoeplinger) [inline mono/utils/mono-time.c]: Hmm, I'm not too familiar with how this method is used but according to the man pages CLOCK_MONOTONIC behaves the same so at least those two codepaths are consistent. This doesn't apply to gettimeofday() of course but not sure on which platforms we fallback to that.

**Pull request merged**